nearcore/src/lib.rs

```diff
@@ -312,16 +313,15 @@ pub fn recompress_storage(home_dir: &Path, opts: RecompressOpts) -> anyhow::Resu
     // (since it’s a command line option) which is why we set home to cwd.
     let cwd = std::env::current_dir()?;
     let dst_opener = NodeStorage::opener(&cwd, &dst_config, None);
+    let dst_opener = dst_opener.expect_archive(archive);
     let dst_path = dst_opener.path();
 
     info!(target: "recompress",
           src = %src_path.display(), dest = %dst_path.display(),
           "Recompressing database");
 
-    let src_store = src_opener
-        .open_in_mode(Mode::ReadOnly)
-        .with_context(|| format!("Opening database at {}", src_opener.path().display()))?
```

Comment:
Not sure what is going on but with the `with_context` the underlying error was not printed to the stderr. I'm putting the relevant info in an info above instead.